src/utils/report-error.ts

```diff
@@ -1,44 +1,20 @@
-import raven from 'lib/raven';
+// This function is used to report errors to Sentry.
+// This uses the `reportError` function from the `window.guardian.modules.sentry` object.
+const reportError = window.guardian.modules.sentry.reportError;
 
-type FrontendError = (Error & { reported?: boolean }) | string;
+type ErrorReportingFunction<T> = (event: T) => void;
 
-const convertError = (err: unknown): Error => {
-	if (err instanceof Error) {
-		return err;
-	}
-	if (typeof err === 'string') {
-		return new Error(err);
-	}
-	return new Error(String(err));
+const wrapWithErrorReporting = <T>(
+	fn: ErrorReportingFunction<T>,
+	tags: Record<string, string> = {},
+): ErrorReportingFunction<T> => {
+	return function (event: T) {
+		try {
+			fn(event);
+		} catch (e) {
+			reportError(e, JSON.stringify(tags));
```

Comment:
We could be a little defensive here and check if `window.guardian.modules.sentry.reportError` is defined before invoking it.

Depending on the timing of how scripts are loaded it is possible (although unlikely) that it does not get initialised before commercial boots.

Reply:
I think we should update this module to wrap `reportError` with an internal function rather than a direct reference at the module level which will only resolve once.

This will allow us to get the reference the function on each invocation and also check if it is defined.